map/map_tests/kmz_unarchive_test.cpp

```diff
@@ -39,3 +40,30 @@ UNIT_TEST(KMZ_UnzipTest)
     TEST_EQUAL(bm.GetScale(), 0, ("KML wrong scale!"));
   }
 }
+
+UNIT_TEST(Multi_KML_KMZ_UnzipTest)
+{
+  std::string const kmzFile = GetPlatform().TestsDataPathForFile("kml_test_data/BACRNKMZ.kmz");
+  auto const filePaths = GetKMLOrGPXFilesPathsToLoad(kmzFile);
+  // Check that doc.kml (could be renamed to docN.kml after unzip) is skipped.
+  std::vector<std::string> expectedFileNames =
+  {
+      "BACRNKMZfilesCampgrounds 26may2022 green and tree icon",
```

Comment:
Can we add a test for zip reader function that returns a list of files in the zip, to find a reliable way of detecting the first file in the root?

We may also try a different approach: try to load every file, even doc.kml, and skip it if it's empty.

Reply:
Resolving as it is discussed here now: https://github.com/organicmaps/organicmaps/pull/6613#discussion_r1412792440